spec.html

```diff
@@ -33734,15 +34812,33 @@ <h1>%TypedArray%.prototype.set ( _array_ [ , _offset_ ] )</h1>
             1. Let _limit_ be _targetByteIndex_ + _targetElementSize_ &times; _srcLength_.
             1. Repeat, while _targetByteIndex_ &lt; _limit_
               1. Let _Pk_ be ! ToString(_k_).
-              1. Let _kNumber_ be ? ToNumber(? Get(_src_, _Pk_)).
+              1. Let _value_ be ? Get(_src_, _Pk_).
+              1. If _target_.[[TypedArrayName]] is `"BigUint64Array"` or `"BigInt64Array"`, let _value_ be ? ToBigInt(_value_).
+              1. Otherwise, let _value_ be ? ToNumber(_value_).
               1. If IsDetachedBuffer(_targetBuffer_) is *true*, throw a *TypeError* exception.
-              1. Perform SetValueInBuffer(_targetBuffer_, _targetByteIndex_, _targetType_, _kNumber_, *true*, `"Unordered"`).
+              1. Perform SetValueInBuffer(_targetBuffer_, _targetByteIndex_, _targetType_, _value_, *true*, `"Unordered"`).
               1. Set _k_ to _k_ + 1.
               1. Set _targetByteIndex_ to _targetByteIndex_ + _targetElementSize_.
             1. Return *undefined*.
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sec-is-copyable-typed-array-types" aoid="IsCopyableTypedArrayTypes">
+          <h1>IsCopyableTypedArrayTypes ( _type_ , _otherType_ )</h1>
+          <p>The abstract operation IsCopyableTypedArrayTypes checks if it is possible to copy elements between TypedArray of types _type_ and _otherType_ :</p>
```

Comment:
I think I didn't understand your question. This function is checking if the type of elements of the 2 TypedArrays are   compatible. By compatible, I mean that there is no mix of `BigInt` and `Number` on `type` and `otherType`. In such case, we return `false` for every combination of `BigInt` and any other type that isn't `BigInt`. In all other cases, we return `true`. In the case you asked, if `type = Int8Array`, it only returns `false` if `otherType = BigInt64 or BigUint64`. This is required to avoid usage of TyperArrays as a way to have implicit cast of `BigInt` and `Number`

Reply:
Ah - so the intention here is to ensure that "both types are bigint" or "both types are number"?

In that case, perhaps it would be cleaner to add an internal slot to all typed arrays like `[[ContentType]]`, and set it to "Number" or "BigInt", and then a) the comparison only needs to compare the ContentType slot - that would also make it extensible to, say, Decimals in the future without a bunch of complex logic?

Reply:
> Ah - so the intention here is to ensure that "both types are bigint" or "both types are number"?
Yes.

> In that case, perhaps it would be cleaner to add an internal slot to all typed arrays like [[ContentType]], and set it to "Number" or "BigInt", and then a) the comparison only needs to compare the ContentType slot - that would also make it extensible to, say, Decimals in the future without a bunch of complex logic?

SGTM. I like this approach.